libraries/6_plugin.rb

```diff
@@ -2,10 +2,20 @@ module GrafanaCookbook
   module Plugin
     module_function
 
-    def available?(name, grafana_cli_bin)
-      cmd = Mixlib::ShellOut.new("#{grafana_cli_bin} plugins list-remote")
-      cmd.run_command
-      !cmd.stdout.split("\n").select { |item| item.include?('id:') && item.match(name) }.empty?
+    def available?(name, plugin_url, grafana_cli_bin)
+      if plugin_url.nil?
+        cmd = Mixlib::ShellOut.new("#{grafana_cli_bin} plugins list-remote")
+        cmd.run_command
+        !cmd.stdout.split("\n").select { |item| item.include?('id:') && item.match(name) }.empty?
+      else
+        response = Net::HTTP.get_response(URI(plugin_url))
```

Comment:
Ok I'm not really sure of the added value of this check anyway. Having something at this Url does not mean that it is a valid plugin or even a valid zip...